### DISCUSSION TITLE
How/whether to write a multi-architecture cask

### DISCUSSION THREAD

**Discussion by vidurb** [Homebrew/discussions#1387] [Casks] [answered]
I'm trying to update the 'Android Studio Preview Canary' cask from [cask-versions](https://github.com/Homebrew/homebrew-cask-versions) to support installing the ARM-specific build on ARM Macs. I don't have a lot of experience with Ruby, but managed to effect the change I wanted by adding if-cases into the Caskfile. You can see the result of this in [my fork](https://github.com/vidurb/homebrew-cask-versions/tree/add-arm-option-android-studio-preview-canary).

The updated Caskfile passes all the checks in the PR template, but I'm unsure as to whether this approach is the one preferred by Homebrew or whether there should be a separate cask altogether for the ARM version of Android Studio Preview Canary. I was unable to find any other casks that had code to differentiate between x86 and ARM architectures, but I wasn't able to find any ARM-specific casks (that weren't for crosscompilation on x86) either. I did find formulae with architecture-specific installation, which is what I referenced to write the code. I was hoping someone from Homebrew could chime in. I understand the advent of Apple Silicon is a sea change for everyone doing macOS packaging and all the quirks might not have been worked out yet.

**Comment by carlocab** [comment, marked as answer]
You might be interested in looking at some of these casks:
```
adobe-creative-cloud
airqmon
appcode
audiogridder-plugin
audiogridder-server
aws-vault
brave-browser
browserosaurus
calmly-writer
clash-for-windows
clion
datagrip
devbook
docker
dosbox-x
drawio
electron
electron-fiddle
eloston-chromium
epic
goland
intellij-idea
intellij-idea-ce
jetbrains-toolbox
katrain
keepassxc
keeweb
logseq
macintoshjs
melonds
miniforge
notion
ocenaudio
ogdesign-eagle
parallels
parallels-virtualization-sdk
phpstorm
pritunl
pycharm
pycharm-ce
racket
ray
remember-the-milk
slack
superhuman
switchhosts
tencent-meeting
terminus
termius
virtualhereserver
visual-studio-code
vlc
webstorm
windows95
zettlr
zoom
zulu
zulufx
```

**Reply by vidurb** [reply]
Thanks.
